packages/rest/src/lib/RequestManager.ts

```diff
@@ -181,6 +191,41 @@ export class RequestManager extends EventEmitter {
 		this.options = { ...DefaultRestOptions, ...options };
 		this.options.offset = Math.max(0, this.options.offset);
 		this.globalRemaining = this.options.globalRequestsPerSecond;
+
+		setInterval(() => {
+			// Only allocate a swept collection if there are listeners
+			const sweptHashes: Collection<string, HashData> | null =
+				this.listenerCount(RESTEvents.HashSweep) > 0 ? new Collection<string, HashData>() : null;
```

Comment:
Listener Count will *always* be 1 here since you're proxying the event through the rest event emitter

Reply:
Looks like this was resolved without changes? (diff shows changes but I can't actually see any)

Reply:
I think the line numbers changed a lot, but it's in the PR diff

Reply:
You're still checking the listener count that will always be 1 in the PR diff